packages/mdl-base-component/adapter.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * A basic component adapter.
+ */
+export const MDLBaseAdapter = {
+  getBoundingClientRect(id) {
+    return this.elements_[id].getBoundingClientRect();
+  },
+
+  setStyles(id, styles) {
+    Object.assign(this.elements_[id].style, styles);
+  },
+
+  forceLayout(id) {
+    return this.elements_[id].offsetWidth;
+  },
+
+  addClass(id, cls) {
+    this.elements_[id].classList.add(cls);
+  },
+
+  removeClass(id, cls) {
+    this.elements_[id].classList.remove(cls);
+  },
+
+  get(...symbolPath) {
```

Comment:
I think we could simplify this; though to clarify it's only passing around a generalized concept of an identifier (in this case, a string), not the actual element itself. The main issue with explicit methods is that there won't be much shared adapter code and also the duplication of methods - addClassToElement1, addClassToElement2, etc. which becomes cumbersome. I'd like to find some happy medium so that each component doesn't need to re-invent the DOM api.